.upstream-tests/test/std/containers/views/mdspan/mdspan.layout.right.ctor/list_init.pass.cpp

```diff
@@ -0,0 +1,34 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <cuda/std/experimental/mdspan>
+#include <cuda/std/cassert>
+
+namespace      stdex = cuda::std::experimental;
+constexpr auto dyn   = stdex::dynamic_extent;
+
+int main(int, char**)
+{
+    {
+        typedef int    data_t ;
+        typedef size_t index_t;
+
+        stdex::layout_right::mapping<stdex::extents<size_t,dyn, dyn>> m{stdex::dextents<size_t,2>{16, 32}};
+
+        static_assert( m.is_exhaustive() == true );
+
+        assert( m.extents().rank()         == 2  );
```

Comment:
[OPTIONAL] `rank()` or `rank_dynamic()` are always `static constexpr` and thus can be checked with `static_assert`.

Reply:
Switched to `static_assert`.